### PR TITLE
3324252610 create bootcamp batch form

### PR DESCRIPTION
**Pull request by samklq**

### CODE REVIEW

File: staff/models/batch.py [hunk 1]
```diff
@@ -63,11 +60,27 @@ def __str__(self):
...
-        if self.objects.count() == 0:
-            # 17 because this will be the next batch number when we launch Learn
-            return 17
-        return self.objects.filter(course__id=course_id).aggregate(models.Max('number'))['number__max'] + 1
+    def next_number(self, course_id, type):
+        course = Course.objects.get(pk=course_id)
+
+        if course.name == Course.CODING_BASICS:
+            if self.basics_objects.count() == 0:
+                # 17 because this will be the first batch number when we launch Learn
```
Comment (rocketkai) [inline staff/models/batch.py:68]: Thank you for these comments

File: staff/tests/models/test_batch_model.py [hunk 2]
```diff
@@ -16,15 +16,66 @@ def test_string_representation(batch_factory):
...
+    first_batch = batch_factory(course=coding_bootcamp_course, type=Batch.FULL_TIME)
+    second_batch = batch_factory(course=coding_bootcamp_course, type=Batch.FULL_TIME)
+    new_capacity = 100
+
+    first_batch.capacity = new_capacity
+    first_batch.save()
+
+    assert first_batch.number == 10
+
+def test_number_assigned_if_new_parttime_bootcamp_batch_record(batch_factory, course_factory):
```
Comment (rocketkai) [inline staff/tests/models/test_batch_model.py:59]: Thank you for these tests

File: student/tests/views/test_registration_views.py [hunk 3]
```diff
@@ -90,30 +73,32 @@ def test_registration_form_renders_batch_on_start_date(batch_factory):
     assert 'id="id_batch_selection-batch_0"' in str(response.content)
 
 
-def test_registration_wizard_form_existing_user(batch_factory, existing_user):
+def test_registration_wizard_form_existing_user(batch_factory, user_factory):
     batch = batch_factory()
+    existing_user = user_factory()
```
Comment (rocketkai) [inline student/tests/views/test_registration_views.py:78]: Nice refactor

**Review by rocketkai — APPROVED** [review]
💪

File: student/tests/views/test_registration_views.py (hunk 3 above)
Comment (samklq) [reply student/tests/views/test_registration_views.py:78]: haha i realise it doesn't work the way i want, so need to take another look

**Pull request merged**